### PR TITLE
Add stopsound command to Glowkit

### PR DESCRIPTION
**Pull request by aramperes**
Adds the stopsound command to Glowkit.
There is currently no implementation of sound categories in Bukkit, so I just removed it from the command.

Glowstone PR coming up.

### CODE REVIEW

**Comment by mastercoms** [comment]
Can we implement sound categories? Do it like the GlowAchievement class in Glowstone's constants, where a SoundCategory is an optional parameter.

**Comment by aramperes** [comment]
I'm not exactly sure what you mean by implementing them. I did add a `Sound.Category` enum in this PR, but as you can see, Sounds are not associated with a category. This is because since 1.9, sounds are no longer associated to a category, and the category has to be specified when playing/stopping the sound. Having a `Sound#getCategory()` getter/field wouldn't make much sense (at least after 1.8).

For reference, the vanilla `sounds.json` file no longer specifies categories (and sneakily looking at MC code, the category is specified when using `playSound()`). This means you can play the `BLOCK_ANVIL_FALL` sound in the `weather` category if you want to.

**Comment by mastercoms** [comment]
Yes but we don't care what Mojang does. We're not supposed to know how they do Minecraft, only how Minecraft itself works. The best way to do this is associate sounds with a category, and allow for overriding the category in when playing sound.

**Comment by aramperes** [comment]
Yes, but how do you get to know what category a sound is assigned to, then? ;)

Edit: found a solution: http://minecraft.gamepedia.com/Sounds.json#Values_as_of_1.8.9
Edit2: Actually, that wouldn't work; these are _sound events_, not the sounds themselves. Sounds don't have a described category, sound events do. That list doesn't match the sound IDs. I guess I'll go by assuming.

**Comment by mastercoms** [comment]
By testing in game and playing each sound with all the other categories muted. Or assuming from each sound name, or looking on a wiki.

**Comment by aramperes** [comment]
I'm going to go with assuming. But, what's even the point with implementing the category? Is there a particular use case you see fit? It's not necessary for `/stopsound`.

**Comment by aramperes** [comment]
@mastercoms does this fit what you said?

**Comment by mastercoms** [comment]
Yes that's great. And thank you for doing all the categories. I only meant for you to add the constructor, since I didn't want you to do all of that boring work. So again, thank you.

It's not necessary for stop sound, but it's for easily playing sounds to their intended category.

File: Paper-API-Patches/0024-Add-stopsound-command.patch [hunk 1]
```diff
@@ -0,0 +1,1291 @@
...
+diff --git a/src/main/java/org/bukkit/command/CommandTarget.java b/src/main/java/org/bukkit/command/CommandTarget.java
+index 74f8fc6..0a33130 100644
+--- a/src/main/java/org/bukkit/command/CommandTarget.java
++++ b/src/main/java/org/bukkit/command/CommandTarget.java
+@@ -261,7 +261,7 @@ public class CommandTarget {
+     /**
+      * Represents the value of a selector argument
+      */
+-    class SelectorValue {
++    public static class SelectorValue {
```
Comment (mastercoms) [inline Paper-API-Patches/0024-Add-stopsound-command.patch:1022]: Could you please explain this change?

Comment (aramperes) [inline Paper-API-Patches/0024-Add-stopsound-command.patch:1022]: @mastercoms 
Quick fix for this: https://github.com/GlowstoneMC/Glowkit/pull/8/files#diff-955e9dd0ca1ee82e7a106f40fa04d1a5R1220

**Pull request merged**